src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -883,20 +883,40 @@ public static int GetProcFSParentPid(int pid)
             {
                 const int invalidPid = -1;
 
-                // read /proc/<pid>/stat
-                // 4th column will contain the ppid, 92 in the example below
-                // ex: 93 (bash) S 92 93 2 4294967295 ...
-                var path = $"/proc/{pid}/stat";
+                // read /proc/<pid>/status
+                // Row beginning with PPid: \d is the parent process id.
+                // This used to check /proc/<pid>/stat but that file was meant
+                // to be a space delimited line but it contains a value which
+                // could contain spaces itself. Using the status file is a lot
+                // simpler because each line contains a record with a simple
+                // label.
+                // https://github.com/PowerShell/PowerShell/issues/17541#issuecomment-1159911577
+                var path = $"/proc/{pid}/status";
                 try
                 {
-                    var stat = System.IO.File.ReadAllText(path);
-                    var parts = stat.Split(' ', 5);
-                    if (parts.Length < 5)
+                    using FileStream fs = File.OpenRead(path);
+                    using StreamReader sr = new(fs);
+                    string line;
+                    while ((line = sr.ReadLine()) != null)
                     {
-                        return invalidPid;
+                        if (!line.StartsWith("PPid:\t", StringComparison.OrdinalIgnoreCase))
+                        {
+                            continue;
+                        }
+
+                        string[] lineSplit = line.Split('\t', 2, StringSplitOptions.RemoveEmptyEntries);
+                        if (lineSplit.Length != 2)
```

Comment:
Typo
```suggestion
                        string[] lineSplit = line.Split('\t', 3, StringSplitOptions.RemoveEmptyEntries);
                        if (lineSplit.Length != 3)
```

Reply:
Why did you change this, I only wanted a max of entries not 3 hence all the checks.

Reply:
`a\tb\t\c` - if we need `b` we should split on 3.

Reply:
That’s not the case here, we only have the header and value. Please revert this commit.

Reply:
Ah, sorry, I forget it is another file.